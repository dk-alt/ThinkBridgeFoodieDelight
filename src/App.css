.navbar {
  width: 100%;
  height: 50px;
  background-color: #162533;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 5px;
  position: sticky;
  top: 0;
  display: flex;
  z-index: 99;
}

body {
  margin: 0;
}

.center-area{
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  padding-top: 10px;
  align-items: center;
  gap: 10px
}

.navbar h3 {
margin: 0;
 color: rgb(255, 255, 255);
 text-shadow: rgb(148, 235, 255) 1px 0 10px;
}

.navbar-action{
  float: right;
}
.navbar-action a {
  text-decoration:  none;
  color: aliceblue;
  padding: 8px;
  text-shadow: rgb(148, 235, 255) 1px 0 10px;
}