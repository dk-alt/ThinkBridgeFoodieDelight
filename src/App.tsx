import Dashboard from './Component/Dashboard/Dashboard'

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css'
import { Home } from './Component/Home/HomePage';
import { Route, Routes } from 'react-router';
import { RestaurantForm } from './Component/RegistrationForm/RestaurantForm';
import { Link } from 'react-router-dom';


const NavBar = ()=> (
  <nav className='navbar' data-testid='navbar'>
    <h3>Foodie Delight Admin</h3>
    <div className='navbar-action'>
      <Link to='/'>Home</Link>
      <Link to='/registration'>Restaurant Registration</Link>
      <Link to='/dashboard'>Restaurent List</Link>
    </div>
  </nav>
);

function App() {
  return (
    <>
      <NavBar />
      <div className='center-area' data-testid='center-area'>
       

        <Routes>
          <Route path="/" Component={Home}/> 
          <Route path="/dashboard" Component={Dashboard}/>
          <Route path="/registration" Component={()=><RestaurantForm onSuccess={()=>{}} isAdd={true} isModal={false}/>}/>
        </Routes>
      </div>
    </>
  )
}

export default App
