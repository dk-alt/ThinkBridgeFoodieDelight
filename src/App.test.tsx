import { render } from '@testing-library/react';

import App from './App';
import { MemoryRouter } from 'react-router-dom';

describe('App',()=>{
    it('App snapshot',()=>{
        const {container} = render(<MemoryRouter><App /></MemoryRouter>)
        expect(container).toMatchSnapshot();
    })

    it('renders navbar and center area',()=>{
        const {queryByTestId} = render(<MemoryRouter><App /></MemoryRouter>)
        expect(queryByTestId('navbar')).toBeInTheDocument();
        expect(queryByTestId('center-area')).toBeInTheDocument();
    })
})