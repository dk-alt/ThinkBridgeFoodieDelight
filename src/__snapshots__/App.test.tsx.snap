// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App snapshot 1`] = `
<div>
  <nav
    class="navbar"
    data-testid="navbar"
  >
    <h3>
      Foodie Delight Admin
    </h3>
    <div
      class="navbar-action"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/registration"
      >
        Restaurant Registration
      </a>
      <a
        href="/dashboard"
      >
        Restaurent List
      </a>
    </div>
  </nav>
  <div
    class="center-area"
    data-testid="center-area"
  >
    <h1>
      Welcome to Admin panel
    </h1>
    <h3>
      Please use the Navigation to Add, View, Update Restaurants!!
    </h3>
    <div
      class="carousel slide"
    >
      <div
        class="carousel-indicators"
      >
        <button
          aria-current="true"
          aria-label="Slide 1"
          class="active"
          data-bs-target=""
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Slide 2"
          data-bs-target=""
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Slide 3"
          data-bs-target=""
          type="button"
        />
      </div>
      <div
        class="carousel-inner"
      >
        <div
          class="active carousel-item"
        >
          <img
            src="https://picsum.photos/id/30/900/500"
            title="First"
          />
          <div
            class="carousel-caption"
          >
            <h3>
              Flavors Unleashed
            </h3>
            <p>
              Discover a world of culinary delights at your fingertips
            </p>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <img
            src="https://picsum.photos/id/42/900/500"
            title="Second"
          />
          <div
            class="carousel-caption"
          >
            <h3>
              Gourmet Guide
            </h3>
            <p>
              Your ultimate companion for exploring exquisite tastes
            </p>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <img
            src="https://picsum.photos/id/63/900/500"
            title="Third"
          />
          <div
            class="carousel-caption"
          >
            <h3>
              Taste Journey
            </h3>
            <p>
              From street food to fine dining, find your next craving
            </p>
          </div>
        </div>
      </div>
      <a
        class="carousel-control-prev"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
          class="carousel-control-prev-icon"
        />
        <span
          class="visually-hidden"
        >
          Previous
        </span>
      </a>
      <a
        class="carousel-control-next"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
          class="carousel-control-next-icon"
        />
        <span
          class="visually-hidden"
        >
          Next
        </span>
      </a>
    </div>
  </div>
</div>
`;
